File: dailyQuestion/2020-06/06-09/golang/solution_mod.go
package main

import "fmt"

func main() {

	num := 12258
	result := translateNum(num)
	fmt.Println(result)

}

func translateNum(num int) int {
	f_2, f_1 := 1, 1
	for num != 0 {
		pre := num % 100
		f := 0
		if pre >= 10 && pre <= 25 {
			f = f_2 + f_1
		} else {
			f = f_1
		}
		f_2 = f_1
		f_1 = f
		num /= 10
	}

	return f_1
}

File: dailyQuestion/2020/2020-06/06-09/python/solution_mod.py
# 动态求余法
def translateNum(num: int) -> int:
    f_1 = f_2 = 1
    while num:
        pre = num % 100
        f = f_1 + f_2 if pre >= 10 and pre <= 25 else f_1
        f_2 = f_1
        f_1 = f
        num = num // 10
    return f_1

if __name__ == '__main__' :
    num = 12258
    result = translateNum(num)
    print(result)
